File: kq-projection-dotnet/kq-projection/Common/Models/Events/AggregateEvent.cs
namespace KQ_Projection.Common.Models.Events;

using KQ_Projection.Common.Enums;
using System.Text.Json.Serialization;

public class AggregateEvent
{
    [JsonPropertyName("gameId")]
    public Guid gameId { get; set; }

    [JsonPropertyName("playerId")]
    public long playerId { get; set; }

    [JsonPropertyName("eventStreamId")]
    public int? eventStreamId { get; set; }

    [JsonPropertyName("revertFromEventStreamId")]
    public int? revertFromEventStreamId { get; set; }

    [JsonPropertyName("eventType")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public EventType eventType { get; set; }

    [JsonPropertyName("initiatedBy")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public EventInitiatedBy initiatedBy { get; set; }

    [JsonPropertyName("timestamp")]
    public long timestamp { get; set; }


    [JsonPropertyName("eventStatus")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public EventStatus eventStatus { get; set; }


    [JsonPropertyName("eventData")]
    public EventData eventData { get; set; }
}

File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/AggregateEvent.java
package com.kafkaquest.kq.common.util.models.events;

import com.kafkaquest.kq.common.util.models.events.eventdata.EventData;
import com.kafkaquest.kq.common.util.models.events.enums.EventInitiatedBy;
import com.kafkaquest.kq.common.util.models.events.enums.EventStatus;
import com.kafkaquest.kq.common.util.models.events.enums.EventType;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.util.UUID;

/**
 * This class represents an Event produced to kafka from Aggregate Service
 */

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AggregateEvent {

    private UUID gameId;

    private Long playerId;

    private Integer eventStreamId;

    private Integer revertFromEventStreamId;

    private EventType eventType;

    private EventInitiatedBy initiatedBy;

    private long timestamp;

    private EventStatus eventStatus;

    private EventData eventData;

}
